### PR TITLE
18833 agm ext calc

### PR DESCRIPTION
**Pull request by jamespaologarcia**
*Issue #:* /bcgov/entity###

Created a method to add up the total number of months in AGM extensions to post the correct value. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-599-ahbbc3ks.web.app

**Comment by severinbeauvais** [comment]
Please update the description template with the ticket number so this PR gets linked in the ticket.

File: src/stores/filingHistoryListStore.ts [hunk 1]
```diff
@@ -38,6 +38,17 @@ export const useFilingHistoryListStore = defineStore('filingHistoryList', {
...
+    /** Returns the total AGM extension duration */
+    getTotalAgmExtensionDuration (state: FilingHistoryListStateIF): number {
+      return state.filings.reduce((totalMonths, filing) => {
+        if (filing.name === FilingTypes.AGM_EXTENSION) {
+          // Add total months
+          totalMonths += filing.data?.agmExtension?.extensionDuration || 0
+        }
+        return totalMonths
+      }, 0)
+    },
```
Comment (severinbeauvais) [inline src/stores/filingHistoryListStore.ts:50]: Very nice 👏

**Review by severinbeauvais — APPROVED** [review]
Please add a unit test that has a few AGM extension filings so that you can verify your new getter. Nice job!

File: src/views/AgmExtension.vue [hunk 2]
```diff
@@ -206,7 +206,7 @@ export default class AgmExtension extends Mixins(CommonMixin, DateMixin,
   @Getter(useRootStore) getUserInfo!: any
   @Getter(useBusinessStore) isGoodStanding!: boolean
   @Getter(useRootStore) isRoleStaff!: boolean
-
+  @Getter(useFilingHistoryListStore) getTotalAgmExtensionDuration!: number
```
Comment (JazzarKarim) [inline src/views/AgmExtension.vue:209]: Super small thing James, we usually leave an empty line between the getters and local variables.

**Review by JazzarKarim — APPROVED** [review]
Looks great James!

**Review by leodube-aot — APPROVED** [review]
LGTM, just the unit test(s) needed

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-599-ahbbc3ks.web.app

File: src/stores/filingHistoryListStore.ts (hunk 1 above)
Comment (severinbeauvais) [reply src/stores/filingHistoryListStore.ts:64]: Still nice!

Which line "cases year as number"? Do you need to do `return +totalMonths` somewhere?

Ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Unary_plus

File: src/views/AgmExtension.vue [hunk 3]
```diff
@@ -430,7 +430,7 @@ export default class AgmExtension extends Mixins(CommonMixin, DateMixin,
         isFirstAgm: this.data.isFirstAgm,
         extReqForAgmYear: this.data.isPrevExtension,
         intendedAgmDate: this.data.intendedAgmDate,
-        totalApprovedExt: this.getTotalAgmExtensionDuration + this.data.extensionDuration,
+        totalApprovedExt: this.getTotalAgmExtensionDuration(Number(this.data.agmYear)) + this.data.extensionDuration,
```
Comment (severinbeauvais) [inline src/views/AgmExtension.vue:433]: Looks good.

Good use of Number vs number btw! 👍

**Review by severinbeauvais — APPROVED** [review]
LGTM!

File: src/views/AgmExtension.vue (hunk 2 above)
Comment (jamespaologarcia) [reply src/views/AgmExtension.vue]: Done

**Pull request merged**